20250412-P-feature-planning.md

```diff
@@ -0,0 +1,182 @@
+and iterated on on-the-fly in a rush to get the feature completed.
+
+The current release cycle structure is ineffective as it forces us to race potentially incomplete ideas out the door.
+
+## Proposal
+I would like to propose several changes to our release process starting immediately.
+
+### Develop Proposals Between Releases
+We launched v1.15 on February 26th. Each of the current goals of this release span proposals introduced largely around
+and since then:
+
+| Proposal | Introduced At                                     | Proposed By |
+| -- |---------------------------------------------------| -- |
+|Actors: Bi-directional streaming | February 17                                       | Josh van Leeuwen |
+| Actors: PubSub | February 26, alternative proposal raised March 25 | Josh van Leeuwen, alternative by Whit Waldo |
+| Multi-App Workflows | March 24                                          | Josh van Leeuwen |
+| Event Streaming | March 24                                          | Cassie Coyle |
+| Workflow Re-runs | March 25, alternative proposal raised April 6     | Whit Waldo, alternative by Josh van Leeuwen |
+
+Despite being raised no more than a couple of weeks ago, many of these proposals have already gained significant traction. 
+However, implementing well-designed flagship features based on a three-week ideation and design timeline is unrealistic
+and isn't the process taken with larger software groups. It shouldn't be done here either. Furthermost, most of these
+issues have minimal community presence outside of the most-active contributors themselves.
+
+#### Proposed Changes
+Now is the time for brainstorming big-picture ideas when contributors take breaks from coding or casual discussions
+spark new inspiration.
+
+Discord frequently features requests for smaller features that could collectively form larger improvements both in terms
+of quality of life and feature additions. Even if these authors don't write original proposals themselves, we should document
+these ideas (in a place the stalebot won't hide them) and solicit community feedback to gauge traction. Regular community
+and meteor calls should highlight these new ideas with balanced commentary from hosts and direct viewers to relevant
+issues so they can chime in themselves.
+
+Designing large-scale features without time pressure soliciting feedback from less-invested contributors can help ensure
+that new capabilities are well-thought-out and either fit into our larger vision or require further refinement.
+
+### Release Prioritization
+Currently, we ideate high-level goals on maintainer and design calls following a release and wait for proposals to be
+submitted, and then largely just move forward with them.
+
+#### Proposed Changes
+I propose a substantial change to our process. Barring hotfixes for intermediate patch releases, only ideas backed by an
```

Comment:
I like this idea, just feeling that the "two months" period should have some level of justification for being that amount of time and not another

Reply:
While we aim for four releases a year, we've been hitting three releases a year, meaning each release is taking on average 4 months or so to complete. So I just picked 2 months as that's half of a 4-month release cycle and feels like a sufficiently long enough period for a "last-minute" idea to start baking and getting public feedback, project alignment, design iteration, etc.